libmscore/realizedharmony.cpp

```diff
@@ -0,0 +1,493 @@
+                        omit |= 1 << 11;
+                        }
+                  // FALLTHROUGH
+            case 9:
+                  if (!(omit & (1 << 9))) {
+                        ret.insert(2 + RANK_MULT*RANK_9TH, tpcInterval(rootTpc, 9, 0));     //maj9
+                        omit |= 1 << 9;
+                        }
+                  // FALLTHROUGH
+            case 7:
+                  if (!(omit & (1 << 7))) {
+                        if (quality == "major")
+                              ret.insert(11 + RANK_MULT*RANK_7TH, tpcInterval(rootTpc, 7, 0));
+                        else if (quality == "diminished")
+                              ret.insert(9 + RANK_MULT*RANK_7TH, tpcInterval(rootTpc, 7, -2));
+                        else //dominant or augmented or minor
+                              ret.insert(10 + RANK_MULT*RANK_7TH, tpcInterval(rootTpc, 7, -1));
+                        }
+                  break;
+            case 6:
+                  if (!(omit & (1 << 6))) {
+                        ret.insert(9 + RANK_MULT*RANK_ADD, tpcInterval(rootTpc, 6, 0));     //maj6
+                        omit |= 1 << 13; //no need to add/alter 6 chords
+                        }
+                  break;
+            case 5:
+                  //omitted third already
+                  break;
+            case 4:
+                  if (!(omit & (1 << 4)))
+                        ret.insert(5 + RANK_MULT*RANK_ADD, tpcInterval(rootTpc, 4, 0));     //p4
+                  break;
+            case 2:
+                  if (!(omit & (1 << 2)))
+                        ret.insert(2 + RANK_MULT*RANK_ADD, tpcInterval(rootTpc, 2, 0));     //maj2
+                  omit |= 1 << 9; //make sure we don't add altered 9 when theres a 2
+                  break;
+            case 69:
+                  ret.insert(9 + RANK_MULT*RANK_ADD, tpcInterval(rootTpc, 6, 0));     //maj6
+                  ret.insert(2 + RANK_MULT*RANK_ADD, tpcInterval(rootTpc, 9, 0));     //maj6
+                  omit = ~0; //no additions/alterations for a 69 chord
+                  break;
+            default:
+                  break;
+            }
+
+      Harmony* next = _harmony->findNext();
+      if (!_literal && next) {
+            //jazz interpretation
+            QString qNext = next->parsedForm()->quality();
+            //pitch from current to next harmony normalized to a range between 0 and 12
+            //add PITCH_DELTA_OCTAVE before modulo so that we can ensure arithmetic mod rather than computer mod
+            int keyTpc = int(next->staff()->key(next->tick())) + 14; //tpc of key (ex. F# major would be Tpc::F_S)
+            int keyTpcMinor = keyTpc + 3; //tpc of relative minor of the key signature (ex. F minor would be Tpc::Ab + 3 = Tpc::F)a
+            int pitchBetween = (tpc2pitch(next->rootTpc()) + PITCH_DELTA_OCTAVE - tpc2pitch(rootTpc)) % PITCH_DELTA_OCTAVE;
+
+            //dont add 9 for chords with altered 5s
```

Comment:
why not?

Reply:
It's to handle diminished chords since it's kind of a bit much in my opinion. Although maybe it could be changed to just not add 9 for diminished chords and minor chords with altered 5s?

Reply:
Why not add ninths to these chords?  Assuming you're adding ninths at all (eg, in a jazz style or whateverr), these are fantastic chords to add ninths to.  Major ninth above root.  And same with dominant sevenths with b5 or #5, major sevenths with #5.

Reply:
Hmm maybe I'm just not super used to using those tones, but for me it seems a little bit too harsh to have a major 9th added for diminished and half diminished chords. But I can take out the code to stop that and maybe we can see? I know you have had more diverse experiences than me so I'll trust your judgement of course :)

Reply:
In a jazz context, a very common voicing for Cmi7b5 would be Eb-Gb-Bb-D, or for Co7, Eb-Gb-A-D (technically, Bbb, not A, of course).  Actually, for fully diminished, it would also be very common to use B instead of A (.  The general rule is, include 2 or 3 notes in the chord, 1 or 2 a whole step above the chord tone.

Reply:
Ah that makes sense. I will re-enable it. I think just in my contexts it wasn't the most appropriate pieces to test the jazz mode with. I was kind of a classically trained theorist turned to jazz so I don't yet understand a decent amount of jazz things out there yet :) Thanks!